sql: TestExplainAnalyzeDebug failed

sql.TestExplainAnalyzeDebug [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3584242&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3584242&tab=artifacts#/) on master @ [1013cdb7bc54db42bc1a62028ad9bd831e995333](https://github.com/cockroachdb/cockroach/commits/1013cdb7bc54db42bc1a62028ad9bd831e995333):


```
=== RUN   TestExplainAnalyzeDebug
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExplainAnalyzeDebug070654654
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestExplainAnalyzeDebug
    explain_bundle_test.go:153: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExplainAnalyzeDebug070654654
--- FAIL: TestExplainAnalyzeDebug (5.24s)
=== RUN   TestExplainAnalyzeDebug/error
    explain_bundle_test.go:93: Get "http://127.196.149.65:39405/_admin/v1/stmtbundle/701731234955591681": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
    --- FAIL: TestExplainAnalyzeDebug/error (3.05s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestExplainAnalyzeDebug PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries RaduBerinde
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainAnalyzeDebug.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/explain_bundle_test.go
+++ b/pkg/sql/explain_bundle_test.go
@@ -22,6 +22,7 @@ import (
 	"sort"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
@@ -157,6 +158,8 @@ CREATE TABLE users(id UUID DEFAULT gen_random_uuid() PRIMARY KEY, promo_id INT R
 // arbitrary number of strings; each string contains one or more filenames
 // separated by a space.
 func checkBundle(t *testing.T, text, tableName string, expectedFiles ...string) {
+	httpClient := httputil.NewClientWithTimeout(30 * time.Second)
+
 	t.Helper()
 	reg := regexp.MustCompile("http://[a-zA-Z0-9.:]*/_admin/v1/stmtbundle/[0-9]*")
 	url := reg.FindString(text)
@@ -164,7 +167,7 @@ func checkBundle(t *testing.T, text, tableName string, expectedFiles ...string) 
 		t.Fatalf("couldn't find URL in response '%s'", text)
 	}
 	// Download the zip to a BytesBuffer.
-	resp, err := httputil.Get(context.Background(), url)
+	resp, err := httpClient.Get(context.Background(), url)
 	if err != nil {
 		t.Fatal(err)
 	}

